### PR TITLE
GH-1286 Reworked 'bindings' endpoint

### PR DESCRIPTION
**Pull request by olegz**

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java [hunk 1]
```diff
@@ -62,25 +53,41 @@ public BindingsEndpoint(List<InputBindingLifecycle> inputBindingLifecycles) {
...
-	public StopEndpoint getStopEndpoint() {
-		return new StopEndpoint();
-	}
-	
-	public StartEndpoint getStartEndpoint() {
-		return new StartEndpoint();
+	@WriteOperation
+	public boolean changeState(@Selector String name, State state) {
+		Assert.notNull(name, "'name' must not be null");
+		Assert.notNull(name, "'state' must not be null");
```
Comment (wilkinsona) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java]: If these asserts ever do anything, there's a bug in Spring Boot. You should only get null if an argument is annotated with `@Nullable`. Your endpoint won't even be called if a required parameter is missing. For example, a `POST` that's missing a `state` will automatically be rejected with a 400 response.

Comment (artembilan) [reply spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java]: Looks like the the second `Assert` must be against `state` param, but `name` again :smile:

Comment (olegz) [reply spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java]: yeah, I am removing both based on Andy's explanation

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java [hunk 2]
```diff
@@ -62,25 +53,41 @@ public BindingsEndpoint(List<InputBindingLifecycle> inputBindingLifecycles) {
...
-	}
-	
-	public StopEndpoint getStopEndpoint() {
-		return new StopEndpoint();
-	}
-	
-	public StartEndpoint getStartEndpoint() {
-		return new StartEndpoint();
+	@WriteOperation
+	public boolean changeState(@Selector String name, State state) {
```
Comment (wilkinsona) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java]: Returning a `boolean` strikes me as a little bit odd, particularly as it'll be `true` unless there's a bug and `state` is `null`.

**Pull request merged**